### PR TITLE
Upgraded What3Words API to v3

### PR DESCRIPTION
**Pull request by saidtezel**

### CODE REVIEW

File: geopy/geocoders/what3words.py [hunk 1]
```diff
@@ -114,14 +107,11 @@ def geocode(
...
 
         if not self._check_query(query):
-            raise exc.GeocoderQueryError(
-                "Search string must be 'word.word.word'"
-            )
+            raise exc.GeocoderQueryError("Search string must be 'word.word.word'")
 
         params = {
-            'addr': query,
-            'lang': lang.lower(),
```
Comment (KostyaEsmukov) [inline geopy/geocoders/what3words.py:123]: `Language` param seems to be missing now. Could you please return it?

File: geopy/geocoders/what3words.py [hunk 2]
```diff
@@ -205,12 +188,13 @@ def reverse(
...
-            'coords': self._coerce_point_to_string(query),
-            'lang': lang.lower(),
-            'key': self.api_key,
+            "coordinates": self._coerce_point_to_string(query),
+            "language": lang.lower(),
+            "key": self.api_key,
         }
 
         url = "?".join((self.reverse_api, urlencode(params)))
+        print(url)
```
Comment (KostyaEsmukov) [inline geopy/geocoders/what3words.py:197]: Could you remove this `print` statement?

File: geopy/geocoders/what3words.py (hunk 1 above)
Comment (saidtezel) [reply geopy/geocoders/what3words.py:123]: In v3, language param is no longer applicable for geocode queries. Please see here: https://developer.what3words.com/public-api/docs#convert-to-coords

File: test/geocoders/what3words.py [hunk 3]
```diff
@@ -80,9 +80,9 @@ async def test_not_exactly_one(self):
         )
 
     async def test_result_language(self):
-        await self.geocode_run(
-            {"query": "piped.gains.jangle", "lang": 'DE'},
-            {"address": 'fortschrittliche.voll.schnitt'},
+        await self.reverse_run(
```
Comment (saidtezel) [inline test/geocoders/what3words.py:83]: Related to API v3 changes for language param requirement, I've modified the `test_result_language` test to use reverse function instead.

**Comment by saidtezel** [comment]
Closing it to reopen a new PR with a new class for What3Words API v3

**Pull request closed without merging**